File: core/src/main/kotlin/com/cameronsh/core/models/message/Message.kt
package com.cameronsh.core.models.message

import com.cameronsh.utils.Id
import java.util.UUID

import com.cameronsh.core.models.task.Task

data class Message(
    val name: String = "Message",
    val task: Task,
    val destination: String
) {
    public val id: UUID = Id.genId()
    val origin: UUID = task.id
}

File: systems/src/engine/iostream/message.rs
use uuid::Uuid;

pub struct Message {
    id: Uuid,
    origin: Uuid,
    destination: Uuid,
}

impl Message {
    fn check_destination(&self, place: Uuid) -> bool {
        return place == self.destination
    }

    fn check_origin(&self, place: Uuid) -> bool {
        return place == self.origin
    }
}
